### PR TITLE
tcgaImport.py/mergeByPlatform.py update

### PR DESCRIPTION
**Pull request by duncan-palmer**
Update tcgaImport SNP6 subclass logic to process SNP6 files as seg and not bed. Add argparse functionality to mergeByPlatform to allow user to input benefactor/parent synId's and add option of specifying particular platforms if only subset need files merged.

### CODE REVIEW

File: mergeByPlatform.py [hunk 1]
```diff
@@ -14,20 +14,33 @@
...
+# Generate string of unique platforms from PLATFORMS array.
+availPlatforms = '\n'.join(set([x[0] for x in PLATFORMS]))
 
-QUERY_STR = "select * from file where benefactorId=='syn2812961'"
+# Argument parser to allow user to indicate with synapse project to merge files from, which project to load the merged file into, 
+# and an optional platform argument if the user only has files of a single platform type. 
+parser = argparse.ArgumentParser()
+parser.add_argument('benefactorId',help='ID of synapse project to merge files from.')
+parser.add_argument('parentId',help='ID of synapse project to add merged file to.')
+parser.add_argument('-p','--platform',help='If merging files of single platform type, add platform name after option. Available platforms' + '\n' +  availPlatforms,type=str)
```
Comment (duncan-palmer) [inline mergeByPlatform.py]: break long strings into multiple lines with ('stasdfasdf'
                                                                   'asdfasdf')

File: mergeByPlatform.py [hunk 2]
```diff
@@ -14,20 +14,33 @@
...
+args = parser.parse_args()
+
+if args.platform:
```
Comment (duncan-palmer) [inline mergeByPlatform.py]: never test for None like this:
do: if args.platform is not None:

File: mergeByPlatform.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
 import hashlib
 from synapseHelpers import query2df, thisCodeInSynapse
 from multiprocessing.dummy  import Pool
+import argparse
 
-PARENT_ID = 'syn4301332'
 PLATFORMS = [('MDA_RPPA_Core', 'RPPA', 'mdanderson.org_PANCAN_MDA_RPPA_Core.RPPA.tsv'),
              ('IlluminaGA_RNASeqV2', 'isoformExp', 'unc.edu_PANCAN_IlluminaGA_RNASeqV2.isoformExp.tsv'),
```
Comment (duncan-palmer) [inline mergeByPlatform.py]: PLATFORMS no longer being used as a constant.  Replace with platform_list or platforms

File: mergeByPlatform.py (hunk 1 above)
Comment (duncan-palmer) [inline mergeByPlatform.py]: Separate multiple parameters over multiple lines if line too long (~80-100 characters.)

Comment (duncan-palmer) [inline mergeByPlatform.py]: I think you can use nargs='*' to accept multiple items in platform

File: mergeByPlatform.py [hunk 4]
```diff
@@ -14,20 +14,33 @@
...
+# Argument parser to allow user to indicate with synapse project to merge files from, which project to load the merged file into, 
+# and an optional platform argument if the user only has files of a single platform type. 
+parser = argparse.ArgumentParser()
+parser.add_argument('benefactorId',help='ID of synapse project to merge files from.')
+parser.add_argument('parentId',help='ID of synapse project to add merged file to.')
+parser.add_argument('-p','--platform',help='If merging files of single platform type, add platform name after option. Available platforms' + '\n' +  availPlatforms,type=str)
+args = parser.parse_args()
+
+if args.platform:
+	PLATFORMS = [x for x in PLATFORMS if x[0] == args.platform]
```
Comment (duncan-palmer) [inline mergeByPlatform.py]: If nargs used.  Go beyond using x[0] and perhaps "x[0] in args.platform"

File: mergeByPlatform.py [hunk 5]
```diff
@@ -48,11 +59,12 @@ def isUptodate(name, files):
...
     filteredMeta = allFiles[(allFiles.platform==platform) & (allFiles.dataSubType==dataSubType) & (allFiles.acronym!='PANCAN')]
     files = mp.map(syn.get, filteredMeta.id)
+
     if isUptodate(name, files):
         print ' is up to date'
         continue
-    
-    if list(set(filteredMeta.fileType))[0] =='bed5':
+
+    if list(set(filteredMeta.fileType))[0] =='seg' or list(set(filteredMeta.fileType))[0] == 'bed5':
```
Comment (duncan-palmer) [inline mergeByPlatform.py]: Not required.  Could do list(set(...) in ['bed5', 'seg']

File: mergeByPlatform.py [hunk 6]
```diff
@@ -48,11 +59,12 @@ def isUptodate(name, files):
...
         dfs = mp.map(lambda f: pd.read_csv(f.path, sep='\t'), files)
         df = pd.concat(dfs, axis=0)
         df.to_csv('/gluster/home/lomberg/tcgaImport/out/'+name, sep='\t', index=False)
```
Comment (duncan-palmer) [inline mergeByPlatform.py]: Should be another parameter

File: mergeByPlatform.py [hunk 7]
```diff
@@ -63,9 +75,12 @@ def isUptodate(name, files):
         df = pd.concat(dfs, axis=1)
         df.to_csv('/gluster/home/lomberg/tcgaImport/out/'+name, sep='\t')
         nFeatures, nSamples = df.shape
+
     print 'Created', name, df.shape
+
```
Comment (duncan-palmer) [inline mergeByPlatform.py]: Probably not needed (according Pep8)

File: tcgaImport.py [hunk 8]
```diff
@@ -32,6 +31,7 @@
 """
 Net query code
 """
+
```
Comment (duncan-palmer) [inline tcgaImport.py]: no?

File: tcgaImport.py [hunk 9]
```diff
@@ -926,55 +929,69 @@ class CGH1x1mImport(TCGASegmentImport):
     }
 
 class SNP6Import(TCGASegmentImport):
+    # Define reference genome.
```
Comment (duncan-palmer) [inline tcgaImport.py]: delete

File: tcgaImport.py [hunk 10]
```diff
@@ -926,55 +929,69 @@ class CGH1x1mImport(TCGASegmentImport):
...
     assembly = 'hg19'
+    # Dictionary which defines dictionaries for each dataSubType, i.e. 'cna'.
+    # The dictionary for each dst maps
```
Comment (duncan-palmer) [inline tcgaImport.py]: what is dst?

File: tcgaImport.py [hunk 11]
```diff
@@ -926,55 +929,69 @@ class CGH1x1mImport(TCGASegmentImport):
...
+            'nameGen' : lambda x : "%s.hg19.cna_nocnv_probecount.seg" % (x)
         }
     }
     
+    # fileScan determines which files from incoming folder to include.
     def fileScan(self, path, dataSubType):
+	# Read in file, which contains one patients data (same barcode).
         with open(path) as handle:
             tmp = pd.read_csv(handle, sep="\t", dtype='object')
+
```
Comment (duncan-palmer) [inline tcgaImport.py]: 86

File: tcgaImport.py [hunk 12]
```diff
@@ -926,55 +929,69 @@ class CGH1x1mImport(TCGASegmentImport):
...
         tmp = tmp.ix[:, ['Sample', 'Chromosome', 'Start', 'End', 'Num_Probes', 'Segment_Mean']]
+
```
Comment (duncan-palmer) [inline tcgaImport.py]: verbotten

File: tcgaImport.py [hunk 13]
```diff
@@ -926,55 +929,69 @@ class CGH1x1mImport(TCGASegmentImport):
...
         self.df = self.df.append(tmp)
         self.df = self.df.ix[:,['Sample', 'Chromosome', 'Start', 'End', 'Num_Probes', 'Segment_Mean']]
 
-
```
Comment (duncan-palmer) [inline tcgaImport.py]: 2 lines between (pep8)

File: tcgaImport.py [hunk 14]
```diff
@@ -926,55 +929,69 @@ class CGH1x1mImport(TCGASegmentImport):
...
             tmp = pd.read_csv(handle, sep="\t", dtype='object')
+
         tmp = tmp.ix[:, ['Sample', 'Chromosome', 'Start', 'End', 'Num_Probes', 'Segment_Mean']]
+
         self.df = self.df.append(tmp)
         self.df = self.df.ix[:,['Sample', 'Chromosome', 'Start', 'End', 'Num_Probes', 'Segment_Mean']]
 
-    
     def fileBuild(self, dataSubType):
+
```
Comment (duncan-palmer) [inline tcgaImport.py]: remove

File: tcgaImport.py [hunk 15]
```diff
@@ -926,55 +929,69 @@ class CGH1x1mImport(TCGASegmentImport):
...
         tmap = self.getTargetMap()
         segFile = "%s/%s.out"  % (self.work_dir, dataSubType)
         self.df["Sample"] = [self.translateUUID(tmap.get(key, key)) for key in self.df["Sample"]]
-        self.df["Chromosome"] = self.df["Chromosome"].apply(correctChrom)
+
+	#SEG files chr name is indicated by number only, i.e. '1' as opposed to 'chr1'. 
+	#self.df["Chromosome"] = self.df["Chromosome"].apply(correctChrom)
+
```
Comment (duncan-palmer) [inline tcgaImport.py]: Removed

File: mergeByPlatform.py [hunk 16]
```diff
@@ -14,20 +15,31 @@
...
+             ('Genome_Wide_SNP_6', 'cna_probecount', 'broad.mit.edu_PANCAN_Genome_Wide_SNP_6.hg19.cna_probecount.seg')]
+             #MSI,Maf
+# Generate string of unique platforms from platforms array.
+availPlatforms = '\n'.join(set([(x[0]+'\t') for x in platforms]))
+# Argument parser to allow user to indicate with synapse project to merge files from, which project to load the merged file into, 
+# and an optional platform argument if the user only has files which are subset of total platforms. 
+parser = argparse.ArgumentParser()
+parser.add_argument('benefactorId',help='ID of synapse project to merge files from.')
+parser.add_argument('parentId',help='ID of synapse project to add merged file to.')
+parser.add_argument('filePath',help='Local filepath to write merged files to.',type=str)
```
Comment (larssono) [inline mergeByPlatform.py]: Set default to '.' so that the parameter is not required.

File: mergeByPlatform.py [hunk 17]
```diff
@@ -14,20 +15,31 @@
...
+parser.add_argument('-p','--platforms',nargs='*',help='If merging subset of platform type, add platform(s) name after option. \
+		    Available platforms' + '\n' +  availPlatforms,type=str)
+args = parser.parse_args()
+if args.platforms is not None:
+	platforms = [x for x in platforms if x[0] in args.platforms]	
+QUERY_STR = "select * from file where benefactorId==" + ("'{0}'").format(args.benefactorId)
```
Comment (larssono) [inline mergeByPlatform.py]: Forgot to mention earlier - this is no longer a constant as well.

File: mergeByPlatform.py [hunk 18]
```diff
@@ -14,20 +15,31 @@
...
 
-QUERY_STR = "select * from file where benefactorId=='syn2812961'"
 
 def isUptodate(name, files):
-    q = syn.chunkedQuery("select id from file where name=='%s' and parentId=='%s'" %(name, PARENT_ID))
+    q = syn.chunkedQuery("select id from file where name=='%s' and parentId=='%s'" %(name, args.parentId))
```
Comment (larssono) [inline mergeByPlatform.py]: Don't use global variables in the function.  PARENT_ID was a constant while args is not constant.

File: mergeByPlatform.py [hunk 19]
```diff
@@ -14,20 +15,31 @@
...
     try:
         id =  q.next()['file.id']
     except StopIteration:
```
Comment (larssono) [inline mergeByPlatform.py]: Lines 42-45 can be replaced with a call to syn._findEntityIdByNameAndParent(name, parent) instead which returns None if the file doesn't exist.

Comment (duncan-palmer) [inline mergeByPlatform.py]: Did you mean lines 43-46?

Comment (larssono) [inline mergeByPlatform.py]: From  q = syn.chunkedQuery
to except StopIteration

You will have to have an additional line to return False if None is returned.

File: mergeByPlatform.py [hunk 20]
```diff
@@ -14,23 +15,31 @@
...
 
-QUERY_STR = "select * from file where benefactorId=='syn2812961'"
 
-def isUptodate(name, files):
-    q = syn.chunkedQuery("select id from file where name=='%s' and parentId=='%s'" %(name, PARENT_ID))
-    try:
-        id =  q.next()['file.id']
-    except StopIteration:
+def isUptodate(name, files,parentId):
+    if not syn._findEntityIdByNameAndParent(name,parentId):
```
Comment (larssono) [inline mergeByPlatform.py]: This won't work.  Where do you store id?  You should test your code before resubmitting.  Also don't test for None by if None.  Use if x is None.

Comment (duncan-palmer) [inline mergeByPlatform.py]: Hey, sorry about that. Working on it now
